sqlite3/src/lib.rs

```diff
@@ -1646,3 +1674,151 @@ fn sqlite3_safety_check_sick_or_ok(db: &sqlite3Inner) -> bool {
+    let db_inner = match db_inner.inner.lock() {
+        Ok(inner) => inner,
+        Err(_) => return SQLITE_MISUSE,
+    };
+
+    // Convert C strings to Rust strings
+    let table_name = match CStr::from_ptr(z_table_name).to_str() {
+        Ok(s) => s,
+        Err(_) => return SQLITE_MISUSE,
+    };
+
+    // Handle database name (can be NULL for main database)
+    let db_name = if z_db_name.is_null() {
+        "main"
+    } else {
+        match CStr::from_ptr(z_db_name).to_str() {
+            Ok(s) => s,
+            Err(_) => return SQLITE_MISUSE,
+        }
+    };
+
+    // For now, we only support the main database
+    if db_name != "main" {
+        rc = SQLITE_ERROR;
+    } else {
+        // Handle column name (can be NULL to just check table existence)
+        if !z_column_name.is_null() {
+            let column_name = match CStr::from_ptr(z_column_name).to_str() {
+                Ok(s) => s,
+                Err(_) => return SQLITE_MISUSE,
+            };
+
+            // Use pragma table_info to get column information
+            let pragma_query = format!("PRAGMA table_info({table_name})");
+            match db_inner.conn.prepare(&pragma_query) {
+                Ok(mut stmt) => {
+                    let mut found_column = false;
+                    loop {
+                        match stmt.step() {
+                            Ok(turso_core::StepResult::Row) => {
+                                let row = stmt.row().unwrap();
+                                let col_name: &str = row.get(1).unwrap(); // name column
+                                if col_name == column_name {
+                                    // Found the column, extract metadata
+                                    let col_type: String = row.get(2).unwrap(); // type column
+                                    let col_notnull: i64 = row.get(3).unwrap(); // notnull column
+                                    let col_pk: i64 = row.get(5).unwrap(); // pk column
+
+                                    z_data_type = CString::new(col_type)
+                                        .expect("CString::new failed")
+                                        .into_raw();
+                                    z_coll_seq = CString::new("BINARY")
+                                        .expect("CString::new failed")
+                                        .into_raw();
+                                    not_null = if col_notnull != 0 { 1 } else { 0 };
+                                    primary_key = if col_pk != 0 { 1 } else { 0 };
+
+                                    // For now, we don't support auto-increment detection
+                                    autoinc = 0;
+
+                                    found_column = true;
+                                    break;
+                                }
+                            }
+                            Ok(turso_core::StepResult::Done) => break,
+                            Ok(_) => continue,
```

Comment:
This is bit weird - although for schema IO shouldn't be needed as it always loaded in memory.

But anyway, I would suggest to reuse/introduce these loops in the lib.rs and just use method which knows what to do here